### PR TITLE
WIP: manuals

### PR DESCRIPTION
**Pull request by Pete-Y-CS**
I haven't added the cross section since I can't convert PDF to HTML with the free tool again today

### CODE REVIEW

**Comment by Pete-Y-CS** [comment]
Still somewhat WIP, the content at the very least needs aligning with that represented by the markdown files in `/manuals`.

However I think it's a reasonable bit to to merge for now as long as  the `Show Manual` buttons aren't too intrusive. There is a weird bug where after closing the modal I can't reopen it, actually

File: scripts/inspectorate_scripts/output.json
Comment (dabreegster) [inline scripts/inspectorate_scripts/output.json:1]: What is output.json?

File: src/routes/route_check/manual/figures/figure3.PNG
Comment (dabreegster) [inline src/routes/route_check/manual/figures/figure3.PNG:1]: Why are we checking this in -- it's not referenced by anything, and it's very Excel centric; we've renamed/combined some of these things

File: src/routes/route_check/manual/content.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+      " The Route Check also introduces a new version of the Junction Assessment Tool (JAT), a framework for assessing pedestrian and cycle movements at junctions. LTN 1/20 introduced a version of the JAT nationally, but that version of the JAT only covered cycle movements at junctions. The Route Check introduces an expanded version of this framework, which also accounts for pedestrian movements. You must assess and score cyclist and pedestrian movements on certain types of junction along the route. You can find more information on this section, including which junction types must be assessed in the JAT, in Chapter 12. ",
+      " Chapter 13 introduces the Results tabs and explains how to interpret results and what might happen after a Route Check has been completed. It also contains information on how the Route Check calculates scores. ",
+    ],
+  },
+  {
+    sectionName: "‌Route Check Structure‌",
```
Comment (dabreegster) [inline src/routes/route_check/manual/content.ts:27]: If you open this in a text editor like vim, you'll see random characters like https://www.compart.com/en/unicode/U+200C on some lines

File: src/routes/route_check/manual/content.ts [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    elements: [
+      " The Route Check, being an Excel-based tool, uses various tabs to categorise and organise the design assessment process. These tabs are divided into input tabs and output tabs. Some input tabs are mandatory, and others are optional. Every tab corresponds to a chapter in this user manual. ",
+      " The input and output tabs are summarised in Figure 2 below, which shows them in the order they appear in the spreadsheet, as well as which tabs are optional depending on whether a street or path is being assessed. Figure 2 – Summary of the Route Check workflow in terms of the tabs in the spreadsheet ",
+      {
+        routeToImage: figure2,
+        number: 2,
+        explanation:
+          "Summary of the Route Check workflow in terms of the tabs in the spreadsheet",
+      },
+      " This chapter explains these tabs in more detail. Input tabs ",
```
Comment (dabreegster) [inline src/routes/route_check/manual/content.ts:37]: image
The HTML text extraction approach is mixing up content and section headings

File: src/routes/area_check/scorecard/+page.svelte [hunk 3]
```diff
@@ -1,9 +1,17 @@
 <script lang="ts">
   import Progress from "./Progress.svelte";
+  import { ManualSectionModal } from "$lib";
+  import { contents } from "../manual/content";
 
   // TODO Include bold in the choices
 </script>
 
+<ManualSectionModal 
+  title={"Area Scorecard Manual"}
```
Comment (dabreegster) [inline src/routes/area_check/scorecard/+page.svelte:10]: Svelte styling: you don't need {} for fixed strings

File: src/routes/area_check/summary/+page.svelte [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <script lang="ts">
-  import { pairs, ExternalLink } from "$lib";
+  import { pairs, ExternalLink} from "$lib";
```
Comment (dabreegster) [inline src/routes/area_check/summary/+page.svelte:2]: There are fmt errors introduced here

File: src/routes/area_check/traffic_mitigation/+page.svelte [hunk 5]
```diff
@@ -2,11 +2,19 @@
   import MajorRoutes from "./MajorRoutes.svelte";
   import MinorRoutes from "./MinorRoutes.svelte";
   import Confirmation from "./Confirmation.svelte";
-  import { TextArea, Select, WarningText } from "govuk-svelte";
+  import { TextArea, Select, WarningText, SecondaryButton } from "govuk-svelte";
   import { state } from "../data";
-  import { YesNo, pairs } from "$lib";
+  import { YesNo, pairs, Modal, ManualSectionModal } from "$lib";
+  import { contents } from "../manual/content";
+  let displayManual = false;
```
Comment (dabreegster) [inline src/routes/area_check/traffic_mitigation/+page.svelte:9]: Unused variable

File: src/lib/manual/ManualPage.svelte
Comment (dabreegster) [inline src/lib/manual/ManualPage.svelte:1]: image
The table of contents is odd. The current section isn't visually highlighted well. I don't think the clickable card component makes sense here; why not just plain HTML numbered/bullet lists + `##x` anchor links?

File: scripts/codegen/README.md
Comment (dabreegster) [inline scripts/codegen/README.md:1]: The nested directory structure doesn't seem too useful, and `scripts/inspectorate_scripts` doesn't say anything about what this does. How about keeping top-level `codegen` as it was and using the `manuals/` directory that's already been started?

**Review by dabreegster — COMMENTED** [review]
I'm confused about https://github.com/acteng/inspectorate_tools/tree/main/manuals vs the new things introduced here -- what approach are we going with? The generated HTML checked in here looks extremely messy compared to the markdown, and after the transformation + rendering, has pieces of different diagrams smushed in:
image
And see other comments where the extracted text has section headings, invisible unicode characters, etc.

I thought we were going with the simpler .md files, editing those to improve the text? In which case to introduce rendering, we could either use an existing md-to-html converter somewhere and just rely on the govuk CSS magic to format everything decently, or if necessary, use a MD parsing library to turn into the JSON format?

I also don't understand the purpose of the custom JSON format for document display -- what does it give us that plain HTML doesn't? If we want to jump to a section of a manual from one page, HTML has the `<a name="section">` thing

**Pull request closed without merging**